tt_fabric/mesh_graph.cpp

```diff
@@ -0,0 +1,302 @@
+            edge.insert({dest_mesh_id, RouterEdge{.port_direction = port_direction, {dest_chip_id}, .weight = 0}});
+        } else {
+            edge[dest_mesh_id].connected_chip_ids.push_back(dest_chip_id);
+        }
+    } else {
+        // Intramesh Connection
+        auto& edge = this->intra_mesh_connectivity_[src_mesh_id][src_chip_id];
+        if (edge.find(dest_chip_id) == edge.end()) {
+            edge.insert({dest_chip_id, RouterEdge{.port_direction = port_direction, {dest_chip_id}, .weight = 0}});
+        } else {
+            edge[dest_mesh_id].connected_chip_ids.push_back(dest_chip_id);
+        }
+    }
+}
+std::unordered_map<chip_id_t, RouterEdge> MeshGraph::get_valid_connections(
+    chip_id_t src_chip_id, std::uint32_t row_size, std::uint32_t num_chips_in_board, FabricType fabric_type) const {
+    std::unordered_map<chip_id_t, RouterEdge> valid_connections;
+    chip_id_t N = src_chip_id - row_size;
+    chip_id_t E = src_chip_id + 1;
+    chip_id_t S = src_chip_id + row_size;
+    chip_id_t W = src_chip_id - 1;
+    if (fabric_type == FabricType::MESH) {
+        if (N >= 0) {
+            valid_connections.insert(
+                {N,
+                 RouterEdge{
+                     .port_direction = RoutingDirection::N,
+                     std::vector<chip_id_t>(this->chip_.num_eth_ports_per_direction, N),
+                     .weight = 0}});
+        }
+        if (E < num_chips_in_board && (E / row_size == src_chip_id / row_size)) {
+            valid_connections.insert(
+                {E,
+                 RouterEdge{
+                     .port_direction = RoutingDirection::E,
+                     std::vector<chip_id_t>(this->chip_.num_eth_ports_per_direction, E),
+                     .weight = 0}});
+        }
+        if (S < num_chips_in_board) {
+            valid_connections.insert(
+                {S,
+                 RouterEdge{
+                     .port_direction = RoutingDirection::S,
+                     std::vector<chip_id_t>(this->chip_.num_eth_ports_per_direction, S),
+                     .weight = 0}});
+        }
+        if (W >= 0 && (W / row_size == src_chip_id / row_size)) {
+            valid_connections.insert(
+                {W,
+                 RouterEdge{
+                     .port_direction = RoutingDirection::W,
+                     std::vector<chip_id_t>(this->chip_.num_eth_ports_per_direction, W),
+                     .weight = 0}});
+        }
+    } else if (fabric_type == FabricType::TORUS) {
+        // TODO: add support
+    }
+    return valid_connections;
+}
+
+void MeshGraph::initialize_from_yaml(const std::string& mesh_graph_desc_file_path) {
+    std::ifstream fdesc(mesh_graph_desc_file_path);
+    TT_FATAL(not fdesc.fail(), "Failed to open file: {}", mesh_graph_desc_file_path);
+
+    YAML::Node yaml = YAML::LoadFile(mesh_graph_desc_file_path);
+
+    TT_FATAL(yaml["Chip"].IsMap(), "MeshGraph: Expecting yaml to define a Chip as a Map");
```

Comment:
A good idea would be to add all of the magic string constants that we expect to see in the yaml file?